edittypes/node_differ.py

```diff
@@ -215,48 +222,47 @@ def get_diff_count(result):
+        is_edit_type_found,wikitext,edit_type = is_edit_type(text,r['type'])
+        if is_edit_type_found:
+            if edit_types.get(edit_type,{}):
+                edit_types[edit_type]['remove'] = edit_types[edit_type].get('remove', 0) + 1
+            else:
+                edit_types[edit_type] = {'remove':1}
+
     for i in result['insert']:
-        sections_affected.add(i["section"])
-    for c in result['change']:
-        sections_affected.add(c['prev']["section"])
+        text = i['text']
+        is_edit_type_found,wikitext,edit_type = is_edit_type(text,i['type'])
+        #check if edit_type in edit types dictionary
+        if is_edit_type_found:
+            if edit_types.get(edit_type,{}):
+                edit_types[edit_type]['insert'] = edit_types[edit_type].get('insert', 0) + 1
+            else:
+                edit_types[edit_type] = {'insert':1}
 
-    edit_types = {}
-    for s in sections_affected:
-        for r in result['remove']:
-            if r["section"] == s:
-                text = r['text']
-                is_edit_type_found,wikitext,edit_type = is_edit_type(text,r['type'])
-                if is_edit_type_found:
-                    if edit_types.get(edit_type,{}):
-                        edit_types[edit_type]['remove'] = edit_types[edit_type].get('remove', 0) + 1
-                    else:
-                        edit_types[edit_type] = {'remove':1}
-
-        for i in result['insert']:
-            if i["section"] == s:
-                text = i['text']
-                is_edit_type_found,wikitext,edit_type = is_edit_type(text,i['type'])
-                #check if edit_type in edit types dictionary
-                if is_edit_type_found:
-                    if edit_types.get(edit_type,{}):
-                        edit_types[edit_type]['insert'] = edit_types[edit_type].get('insert', 0) + 1
-                    else:
-                        edit_types[edit_type] = {'insert':1}
-
-        for c in result['change']:
-            
-            if c["prev"]["section"] == s:
-                if c['prev']['type'] == c['curr']['type']:
-                    is_edit_type_found,edit_type = is_change_in_edit_type(c['prev']['text'],c['curr']['text'],c['prev']['type'])
+    for c in result['change']:
+        if c['prev']['type'] == c['curr']['type']:
+            is_edit_type_found,edit_type = is_change_in_edit_type(c['prev']['text'],c['curr']['text'],c['prev']['type'])
+            #check if edit_type in edit types dictionary
+            if is_edit_type_found:
+                if edit_types.get(edit_type,{}):
+                    edit_types[edit_type]['change'] = edit_types[edit_type].get('change', 0) + 1
+                else:
+                    edit_types[edit_type] = {'change':1}
+
+
+    for m in result['move']:
+        if m['prev']['type'] == m['curr']['type']:
+            is_edit_type_found,edit_type = is_change_in_edit_type(m['prev']['text'],m['curr']['text'],m['prev']['type'])
```

Comment:
We won't want to run the change check -- with a move, there shouldn't be any change in the content (it's literally the exact same node). At a later point, I might consider detecting "fuzzy" moves -- e.g., almost the exact same content found in different places -- but let's not deal with that yet. To help check, maybe also write a quick test case for moves -- e.g., remove a category from the end of the document and add it to the top.

Reply:
an edit to my previous thought -- this might be slightly more involved than I thought. we need to run something like the is_change_in_edit_type function because we need to distinguish between the different tags -- e.g., is a Tag a reference or table etc. Though we should discuss because maybe this is a nudge to move that logic to the tree differ too (like we did for media vs. category vs. wikilink)